source/digits_hits/src/GateToRoot.cc

```diff
@@ -621,15 +633,16 @@ void GateToRoot::RecordBeginOfEvent(const G4Event *evt) {
 
     theCRData.photon1_phantom_Rayleigh = 0;
     theCRData.photon2_phantom_Rayleigh = 0;
+    theCRData.photon3_phantom_Rayleigh = 0;
     theCRData.photon1_phantom_compton = 0;
     theCRData.photon2_phantom_compton = 0;
-    strcpy(theCRData.theComptonVolumeName1, G4String("NULL").c_str());
-    strcpy(theCRData.theComptonVolumeName2, G4String("NULL").c_str());
-    strcpy(theCRData.theRayleighVolumeName1, G4String("NULL").c_str());
-    strcpy(theCRData.theRayleighVolumeName2, G4String("NULL").c_str());
-    m_ionDecayPos = G4ThreeVector(0., 0., 0.);
-    m_positronGenerationPos = G4ThreeVector(0., 0., 0.);
-    m_positronAnnihilPos = G4ThreeVector(0., 0., 0.);
```

Comment:
In my opinion m_ionDecayPos, m_positronGenerationPos and m_positronAnnihilPos should be present in the method GateToRoot::RecordBeginOfEvent. They should be reseted for each event.

In my opinion, you accidentally deleted these lines - this won't show up as a compile-time error, but it's a runtime error.

Please compare your code with https://github.com/OpenGATE/Gate/blob/develop/source/digits_hits/src/GateToRoot.cc#L607